module/zfs/spa.c

```diff
@@ -3098,6 +3098,9 @@ spa_load_best(spa_t *spa, spa_load_state_t state, int mosconfig,
 
 	if (state == SPA_LOAD_RECOVER) {
 		ASSERT3P(loadinfo, ==, NULL);
+		if (config && (rewind_error == 0)) {
+			nvlist_free(config);
+		}
```

Comment:
nvlist_free() can safely accept NULL as an argument, so what about this instead as a simplification.  It makes it clear that either the generated config is either stored with the spa or freed.

```diff
diff --git a/module/zfs/spa.c b/module/zfs/spa.c
index da238a3..7cf2350 100644
--- a/module/zfs/spa.c
+++ b/module/zfs/spa.c
@@ -3095,6 +3095,8 @@ spa_load_best(spa_t *spa, spa_load_state_t state, int mosconfig,
 
        if (config && (rewind_error || state != SPA_LOAD_RECOVER))
                spa_config_set(spa, config);
+       else
+               nvlist_free(config)
 
        if (state == SPA_LOAD_RECOVER) {
                ASSERT3P(loadinfo, ==, NULL);
```